data/talkactions/scripts/create_item.lua

```diff
@@ -8,10 +8,17 @@ function onSay(player, words, param)
 	end
 
 	local split = param:split(",")
+	if not split[1] then
+		player:sendCancelMessage("You need to specify an item name or id.")
+		return false
+	end
 
 	local itemType = ItemType(split[1])
 	if itemType:getId() == 0 then
-		itemType = ItemType(tonumber(split[1]))
+		if tonumber(split[1]) then
+			itemType = ItemType(tonumber(split[1]))
+		end
+
 		if itemType:getId() == 0 then
```

Comment:
Yes, it is quite unnecessary to perform the check although we know for sure it will fail if above condition passes. Although, I felt it was better than creating variables to check for success (or repaeting the cancel message).